src/app/ButtonWrapper/ButtonWrapper.tsx

```diff
@@ -0,0 +1,65 @@
+import { CustomerInitializeOptions, CheckoutSelectors, CustomerRequestOptions } from '@bigcommerce/checkout-sdk';
+import { noop } from 'lodash';
+import React from 'react';
+import { CheckoutContextProps, withCheckout } from '../checkout';
+import CheckoutButtonList from '../customer/CheckoutButtonList';
+
+interface ButtonWrapperProps {
+    checkEmbeddedSupport(methodIds: string[]): void;
+    onUnhandledError?(error: Error): void;
+}
+
+interface WithButtonWrapperProps {
+    checkoutButtonIds: string[];
+    isInitializing: boolean;
+    providerWithCustomCheckout?: string;
+    initializeCustomer(options: CustomerInitializeOptions): Promise<CheckoutSelectors>;
+    deinitializeCustomer(options: CustomerRequestOptions): Promise<CheckoutSelectors>; 
+}
+
+function ButtonWrapper({
+    checkoutButtonIds,
+    isInitializing,
+    checkEmbeddedSupport,
+    onUnhandledError = noop,
+    initializeCustomer,
+    deinitializeCustomer
+}: ButtonWrapperProps & WithButtonWrapperProps) {
+    return (
+        <>
+            {/* <div className='wallet-button-header'>Express checkout</div> */}
+            <div className='wallet-button-container'>
+                <CheckoutButtonList
+                    checkEmbeddedSupport={ checkEmbeddedSupport }
+                    deinitialize={ deinitializeCustomer }
+                    initialize={ initializeCustomer }
+                    isInitializing={ isInitializing }
+                    methodIds={ checkoutButtonIds }
+                    onError={ onUnhandledError }
+                />
+            </div>
+            <div className='or-normal'><span>OR</span></div>
+        </>
+    );
+}
+
+export function mapToButtonComponentProps(
+    { checkoutService, checkoutState }: CheckoutContextProps
+): WithButtonWrapperProps {
+    const {
+        data: { getConfig },
+        statuses: { isInitializingCustomer },
+    } = checkoutState;
+
+    const config = getConfig();
+
+    return {
+        checkoutButtonIds: config?.checkoutSettings.remoteCheckoutProviders as string[],
```

Comment:
That list shouldn't really matter. For example, we could make a call to `/api/storefront/payments`, collect the method IDs and put them here instead.

We should probably call `initializeCustomer` in the `Customer` component instead of `CheckoutButtonList`, considering it may or may not insert wallet button. If we do that, the call should have visibility on all the child elements (email field, button container etc...) within the customer step and therefore can handle both cases.

Reply:
I used the `remoteCheckoutProviders` since that's what I can test with in my local environment. As David stated it should not matter, since the `<CheckoutButtonList />` component should take a list of `checkoutButtonIds`/`methodIds` and initialise that accordingly.

Reply:
thanks @davidchin 

> That list shouldn't really matter. For example, we could make a call to /api/storefront/payments, collect the method IDs and put them here instead.

I am not sure we can do that, because list in there returns regular wallet providers, while we need button providers. I believe it is the reason why customer strategy pulls providers with remote provider interface `RemoteCheckoutProviderInterface`. Nevertheless, for the cart page wallet buttons, we use `StorefrontCheckoutButtonProviderInterface`

Reply:
@animesh1987 I agree that it does not really metter on where to pull it from. However, If we assume that we need to rely on `RemoteCheckoutProviderInterface` then for either top button or customer step we will use the same list of providers (otherwise we will overcomplex the solution even on backend). But since some providers are working for the customer step and for the top checkout place, it makes me think that there possible conflicts in using customer strategy in there

Reply:
Per our discussion with @animesh1987 cc @davidchin 

Animesh plans on moving the component inside the customer step. In that case almost all our doubts are resolved because Customer step = Customer strategy. It will intuitive and straightforward for developers to figure it out how it works and what it does. Since bolt does not a button and intercepts the actual login - until it exists in the scope of the customer step we should not have any issues or conflicts with it.

However, we briefly touched the question of collapsing customer step along with the buttons in the case a customer is logged in. 

Such behaviour may reduce usability and value of the buttons feature, because buttons should provide a way to pay (not to login), if we collapse these for all logged in users, it will work only for guests then. It is someting @nkostohryz @bc-charlesho need to align on.

Reply:
> I am not sure we can do that

@BC-ACherednichenko It should be fine. 🙂 The reason why we read from `remoteCheckoutProviders` is to save an extra network call, not as a way to do filtering. For those that don’t require initialisation, it’d just fall back to the default, which is essentially a no-op.

`StorefrontCheckoutButtonProviderInterface` exists for a different reason due to the way BCApp conditionally injects Twig template. It’s done that way because we used to have the cart buttons implemented in BCApp. Now we no longer do, so with a bit of refactoring we don’t actually need it.